driver-core/src/main/com/mongodb/internal/operation/QueryBatchCursor.java

```diff
@@ -156,11 +169,18 @@ public boolean hasNext() {
 
     @Override
     public List<T> next() {
-        if (closed) {
+        if (!closeableState.startOperation()) {
             throw new IllegalStateException("Iterator has been closed");
```

Comment:
Same comment as before (error message is wierdly different).

Perhaps a loan method could dry this logic up:

Reply:
We have different error messages depending on what methods of the `QueryBatchCursor` detected that it is closed. However, I tried removing the repetition by moving the code that throws `IllegalStateException` to `ResourceManager.execute`, which is essentially the `inOperation` you suggested.

Since most of your concerns are similar, I will not reply to all the threads to avoid duplication. What do you think about the changed code?

Reply:
Like the changes :+1: